src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java

```diff
@@ -284,12 +363,32 @@ private static void deleteFolder(File directory) {
 	@Override
 	public List<RevCommit> getCommits() {
 		List<RevCommit> commits = new ArrayList<RevCommit>();
-		for (Ref branch : getAllRefs()) {
+		for (Ref branch : getOnlyRemoteRefs()) {
+			/* @issue: All branches will be created in separate file system
+			 * folders for this method's loop. How can this be prevented?
+			 *
+			 * @alternative: remove this method completely,
+			 * fetching commits from all branches is not sensible!
```

Comment:
Yes, I think it's a bad idea to not fetch all commits of all branches. 
[alternative] We could check wether the JIRA issue key is part of the branch name and - if so - only use the commits from this branch. [/alternative]

Reply:
Thanks for your thoughts, I wil add the alternative to the code.